src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -93,6 +93,51 @@ internal static class GoToDefinitionHelpers
             return definitions.ToImmutableAndFree();
         }
 
+        public static ImmutableArray<DefinitionItem> GetOverridenDefinitions(
+            ISymbol symbol,
+            Project project,
+            bool thirdPartyNavigationAllowed,
+            CancellationToken cancellationToken)
+        {
+            if (symbol is IMethodSymbol method && method.IsOverride)
+            {
+                return GetDefinitions(method.OverriddenMethod, project, thirdPartyNavigationAllowed, cancellationToken);
+            }
+
+            return ImmutableArray.Create<DefinitionItem>();
+        }
+
+        public static ImmutableArray<DefinitionItem> GetImplementedDefinitions(
+            ISymbol symbol,
+            Project project,
+            bool thirdPartyNavigationAllowed,
+            CancellationToken cancellationToken)
+        {
+            ArrayBuilder<DefinitionItem> definitions = null;
+            if (symbol is IMethodSymbol method)
+            {
+                if (method.ContainingSymbol is INamedTypeSymbol containingType)
+                {
+                    foreach (var @interface in containingType.AllInterfaces)
+                    {
+                        foreach (var member in @interface.GetMembers())
+                        {
+                            var implementation = containingType.FindImplementationForInterfaceMember(member);
+                            if (implementation is null || !implementation.ContainingSymbol.Equals(containingType))
```

Comment:
i don't see how the right side of the || could ever happen.

Reply:
actually, to be more accurate, it's not clear ot me why you're doing that check at all.   Shouldn't you be looking to see if 'method' is an implementation of this interface member?    i don't see that check anywhere.

Reply:
this can also be written in a way that doesn't depend on the type being a method.